### ISSUE TITLE
`section-underline-matches-section-length` (`D409`) should warn on all titles instead of a list of known ones

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#9727]
it seems that this rule only cares about specific titles ("Example", "Parameters", "Raises", "Returns") but i don't see why it can't just check every title

```py
def foo(): # error
    """
    example:
    -----------
    asdf
    """

def bar(): # no error
    """
    asdf:
    --------
    asdf
    """
```

https://play.ruff.rs/a10dd179-6047-43c4-97b1-5083e68008a1
